src/components/UserSelector.tsx

```diff
@@ -38,10 +132,20 @@ export const UserSelector: React.FC<Props> = ({
     return () => {
       document.removeEventListener('click', handleDocumentClick);
     };
-    // we don't want to listening for outside clicks
-    // when the Dopdown is closed
   }, [expanded]);
 
+  if (loading) {
+    return (
+      <div data-cy="UserSelector">
+        <Loader />
+      </div>
+    );
+  }
+
+  if (error) {
+    return <div data-cy="UserSelector">{error}</div>;
```

Comment:
Consider providing a more user-friendly error message or using a notification component to enhance user experience.